src/Magnum/GL/Buffer.cpp

```diff
@@ -468,11 +468,17 @@ void Buffer::getParameterImplementationDSA(const GLenum value, GLint* const data
 }
 #endif
 
-#ifndef MAGNUM_TARGET_GLES
+#if !defined(MAGNUM_TARGET_GLES) || (defined(MAGNUM_TARGET_WEBGL) && !defined(MAGNUM_TARGET_GLES2))
 void Buffer::getSubDataImplementationDefault(const GLintptr offset, const GLsizeiptr size, GLvoid* const data) {
+    #if !defined(CORRADE_TARGET_EMPSCRIPTEN) || (__EMSCRIPTEN_major__*10000 + __EMSCRIPTEN_minor__*100 + __EMSCRIPTEN_tiny__ >= 20017)
     glGetBufferSubData(GLenum(bindSomewhereInternal(_targetHint)), offset, size, data);
+    #else
+    CORRADE_ASSERT_UNREACHABLE("GL::Buffer::subData(): Buffer data queries in WebGL 2.0 possible only since Emscripten 2.0.17", );
```

Comment:
As we discussed on Gitter, i'd make the whole thing fail at compile time, and not at runtime, to prevent an accidental footgun. So wrap this and also `Buffer::subData()` in `#if __EMSCRIPTEN_major__*...`.

Hmm but that means all test code using `DebugTools::bufferSubData()` now has to do `#if __EMSCRIPTEN_major__*...` as well, umm... OTOH, it'd have to do that anyway, to prevent a runtime assert, so I guess that's fine. Yeah, so make this function disabled at compile time instead.

Reply:
I was thinking... couldn't we fall back to using `EM_ASM` to call `getBufferSubData()` function on older Emscripten versions? There'd be some runtime overhead on those versions, but it's much better than not being able to read buffers at all. And it'd remove the need for ugly version checks around the two functions as well as in all tests.

Reply:
Realistically, is anybody using such old Emscripten anymore? (Except me on the CI, which I should upgrade anyway.)

If ancient Emscriptens would be regularly used by people because they're in LTS Linux distro repos (which is the case with old GCC and old CMake), then it would make sense to add some fallback path, but I don't think anybody really does that.

I'd say -- let's upgrade the WebGL 2 build on the CI to 2.0.17 to not have to ifdef each and every test (can you do that in the PR?) but keep the API in `GL::Buffer` ifdef'd to not cause a linker failure for *users* on older versions. I don't expect regular users to build tests as well, so this should be fine. Then support for < 2.0.17 is just on a best-effort basis (we aim to support but we don't explicitly test for it). To keep at least some variety in the testing, the WebGL 1 CI build would stay on 1.39 for now and the next upgrade round (in a year or so?) would bump WebGL 2 build to 3.0.x and WebGL 1 to 2.0.x.

Hm, now I remember we still have the compatibility code for pre-1.39 event handling behavior in the Application classes, right? Probably a good time to drop that now :)

Reply:
I upgraded to 2.0.17 for the emscripten-webgl2 job but it's failing with very funny errors. Not sure if we actually hit a compiler error there, but I'll try to build with that exact version locally.